### PR TITLE
validatePkgConfig: init

### PR DESCRIPTION
**Pull request by danieldk**
###### Motivation for this change

This adds the `validatePkgConfig` hook, which can be used to validate
pkg-config files in the output(s). Currently, this will just run
`pkg-config --validate` on all `.pc` files, capturing errors such as
the issue that was fixed in #87789.
    
The hook could be extended in the future with more fine-grained
checks.

I have also added a commit which enables this hook for the MKL derivation as an example.

This is the first time I touched the hooks at all, so I hope I didn't do anything bad ;).

@bhipple 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by danieldk** [comment]
Seems like this PR sunk down the list ;). Any comments @bhipple ?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -501,6 +501,10 @@ in
 
   iconConvTools = callPackage ../build-support/icon-conv-tools {};
 
+  validatePkgConfig = makeSetupHook
+    { deps = [ findutils pkgconfig ]; }
```
Comment (Ericson2314) [inline pkgs/top-level/all-packages.nix:505]: Can you add a `name` parameter here?

I have been meaning to make that required at some point.

Comment (danieldk) [reply pkgs/top-level/all-packages.nix]: Thanks! Fixed.

**Review by Ericson2314 — APPROVED** [review]

**Review by bhipple — APPROVED** [review]

**Comment by bhipple** [comment]
Result of `nixpkgs-review pr 87866` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
<br>- clmagma
<br>- mkl
<br>- python27Packages.mkl-service
<br>- python37Packages.mkl-service
<br>- python38Packages.mkl-service
<br>- validatePkgConfig
</details>

**Review by bhipple — APPROVED** [review]

**Pull request merged**